ginepro/src/service_definition.rs

```diff
@@ -37,10 +37,10 @@ impl ServiceDefinition {
     }
 }
 
-impl TryFrom<(&str, u16)> for ServiceDefinition {
+impl<T: ToString> TryFrom<(T, u16)> for ServiceDefinition {
     type Error = anyhow::Error;
 
-    fn try_from((hostname, port): (&str, u16)) -> Result<Self, Self::Error> {
+    fn try_from((hostname, port): (T, u16)) -> Result<Self, Self::Error> {
         Self::from_parts(hostname, port)
     }
 }
```

Comment:
Should we use a generic parameter? This might cause us issue if we want to add more implementations later on.

Reply:
We already do for public `from_parts` API so any drawbacks have already been accepted. The `TryFrom` is just a helper for that essentially, so it makes sense to keep the signatures consistent.

Reply:
Mmmm it's not quite the same - you can always add a new method (e.g. `from_boxed_parts`), but you can't add more generic implementations of `TryFrom<(T, u16)>` with different bounds on `T`.

Reply:
:thinking: yes I suppose so. Well we could just implement for `host: String` which was the motivating factor.

Reply:
That's what I would suggest 😁